test/test_v2/test_common.c

```diff
@@ -204,16 +258,92 @@ void free_test_params(test_params *params) {
     }                                 
 } 
 
-void pmixt_pre_init(int argc, char **argv, test_params *params) {
+void free_validation_params(validation_params *val_params) { 
+    //placeholder for if/when we have something that needs to be freed
+    ;
+} 
+
+void set_client_argv(test_params *params, char ***argv)
+{
+    pmix_argv_append_nosize(argv, params->binary);
+    pmix_argv_append_nosize(argv, "-n");
+    if (NULL == params->np) {
+        pmix_argv_append_nosize(argv, "1");
+    } else {
+        pmix_argv_append_nosize(argv, params->np);
+    }
+    if( params->verbose ){
+        pmix_argv_append_nosize(argv, "-v");
+    }
+    if (NULL != params->prefix) {
+        pmix_argv_append_nosize(argv, "-o");
+        pmix_argv_append_nosize(argv, params->prefix);
+    }
+    if (params->nonblocking) {
+        pmix_argv_append_nosize(argv, "-nb");
+    }
+    if (params->collect) {
+        pmix_argv_append_nosize(argv, "-c");
+    }
+    if (params->collect_bad) {
+        pmix_argv_append_nosize(argv, "--collect-corrupt");
+    }
+}
 
+void pmixt_pre_init(int argc, char **argv, test_params *params, validation_params *v_params) {
+
+    size_t v_size = 1;
     init_test_params(params);
+    init_validation_params(v_params);
+    /* sizeof checks for debugging - only uncomment if there's a problem (not normal verbose output)
+    TEST_VERBOSE(("sizeof(*v_params) = %ld, sizeof(v_params.version) = %ld, sizeof(v_params.validate_params) = %ld, \
+            \nsizeof(v_params.check_pmix_rank) = %ld, sizeof(v_params.pmix_rank) = %ld, \
+            \nsizeof(v_params.check_pmix_job_size) = %ld, sizeof(v_params.pmix_job_size = %ld,\
+            \nv_params.pmix_rank = %u, v_params.pmix_job_size = %u\n",
+            sizeof(*v_params),
+            sizeof(v_params->version),
+            sizeof(v_params->validate_params),
+            sizeof(v_params->check_pmix_rank),
+            sizeof(v_params->pmix_rank),
+            sizeof(v_params->check_pmix_job_size),
+            sizeof(v_params->pmix_job_size),
+            (v_params->pmix_rank),
+            (v_params->pmix_job_size)
+             ));
+    */
     parse_cmd(argc, argv, params);
+    //parse_cmd_client(argc, argv, params, v_params);
+    if (params->validate_params) {
+        const char *vptr = pmixt_decode(v_params_ascii_str, &v_size);
```

Comment:
```suggestion
        int count = pmixt_decode(v_params_ascii_str, (void*)&v_params, sizeof(v_params), size-need-to-match);
        assert(count == sizeof(*v_params));
        /* 
        count == -1 if the buffer overruns
        otherwise it reflects the unpacked sizes
        */
```

Reply:
should be fixed.